server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/objectstore/BlobStoreChoosingModule.java

```diff
@@ -0,0 +1,133 @@
+        private final HashBlobId.Factory blobIdFactory;
+
+        @Inject
+        CassandraBlobStoreFactory(Session session, CassandraConfiguration configuration, HashBlobId.Factory blobIdFactory) {
+            this.session = session;
+            this.configuration = configuration;
+            this.blobIdFactory = blobIdFactory;
+        }
+
+        @Override
+        public BlobStore get() {
+            return new CassandraBlobsDAO(
+                session,
+                configuration,
+                blobIdFactory);
+        }
+    }
+
+    static class SwiftBlobStoreFactory implements BlobStoreFactory {
+        private final ObjectStorageBlobsDAOProvider blobsDAOProvider;
+
+        @Inject
+        SwiftBlobStoreFactory(ObjectStorageBlobsDAOProvider blobsDAOProvider) {
+            this.blobsDAOProvider = blobsDAOProvider;
+        }
+
+        @Override
+        public BlobStore get() {
+            return blobsDAOProvider.get();
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BlobStoreChoosingModule.class);
+
+    static final String BLOBSTORE_CONFIGURATION_NAME = "objectstore";
+
+    @Override
+    protected void configure() {
+        bind(SwiftBlobStoreFactory.class).in(Scopes.SINGLETON);
+        bind(PayloadCodec.class).toProvider(PayloadCodecProvider.class).in(Scopes.SINGLETON);
+
+        bind(CassandraBlobStoreFactory.class).in(Scopes.SINGLETON);
+        Multibinder<CassandraModule> cassandraDataDefinitions = Multibinder.newSetBinder(binder(), CassandraModule.class);
+        cassandraDataDefinitions.addBinding().toInstance(CassandraBlobModule.MODULE);
+    }
+
+    @VisibleForTesting
+    @Provides
+    @Singleton
+    BlobStoreFactory provideBlobStoreFactory(PropertiesProvider propertiesProvider,
+                                             CassandraBlobStoreFactory cassandraBlobStoreFactory,
+                                             SwiftBlobStoreFactory swiftBlobStoreFactory) throws ConfigurationException {
+        try {
+            Configuration configuration = propertiesProvider.getConfiguration(BLOBSTORE_CONFIGURATION_NAME);
+            BlobStoreChoosingConfiguration choosingConfiguration = BlobStoreChoosingConfiguration.from(configuration);
+            switch (choosingConfiguration.getImplementation()) {
+                case SWIFT:
+                    return swiftBlobStoreFactory;
+                case CASSANDRA:
+                    return cassandraBlobStoreFactory;
+                default:
+                    throw new RuntimeException(String.format("can not get the right blobstore provider with configuration %s",
+                        choosingConfiguration.toString()));
+            }
+        } catch (FileNotFoundException e) {
+            LOGGER.error("Could not find " + BLOBSTORE_CONFIGURATION_NAME + " configuration file, using cassandra blobstore as the default");
+            return cassandraBlobStoreFactory;
```

Comment:
I'm not convinced of this choice, why shouldn't we fail?

Reply:
Because James should start with a minimum configuration files.

This choice allow to start a James server without `objectstore.properties` thus is valuable.

Reply:
then it should be a `.warn`, no?